.github/workflows/dco.yml

```diff
@@ -0,0 +1,39 @@
+name: Test Developer Certificate of Origin
+
+on:
+  pull_request:
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    name: Test Developer Certificate of Origin
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+          persist-credentials: false
+      - name: Determine branch name
+        run: |
+          BRANCH="${GITHUB_BASE_REF#refs/heads/}"
+          echo "Checking DCO for every commit on branch ${BRANCH}"
+          echo "BRANCH=${BRANCH}" >> ${GITHUB_ENV}
+      - name: Test DCO for every commit
+        run: |
+          RET=0
+          # this will not work properly if what we are testing is not a pull request
```

Comment:
Maybe change the logic to return 1 by default? So init RET to 1 and only set RET=0 if all commits match? THat way we only succeed actively if all is fine, and any unexpected thing (like an empty list) will fail.